src/pygama/lgdo/lh5_store.py

```
@@ -370,35 +383,33 @@ def read_object(self, name, lh5_file, start_row=0, n_rows=sys.maxsize, idx=None,
                 obj_buf.resize(do_warn=True)
                 # set (write) loc to end of tree
                 obj_buf.loc = obj_buf_start+n_rows_read
-                #check attributes
+                # check attributes
                 if set(obj_buf.attrs.keys()) != set(attrs.keys()):
-                    print('warning: attrs mismatch')
-                    print('obj_buf.attrs:', obj_buf.attrs)
-                    print('h5f['+name+'].attrs:', attrs)
+                    log.warning(
```

Is this a warning or an error?

---

we should raise an exception here too.